### ISSUE TITLE
Crash. Reproduced once on Samsung Galaxy S4 SGH-i337

### ISSUE THREAD

**Issue by DmitriyZaitsev** [koral--/android-gif-drawable#140]
See stacktrace below:
```java
E/AndroidRuntime(15676): 	at com.b.a.l.uncaughtException(Unknown Source)
E/AndroidRuntime(15676): 	at pl.droidsonroids.gif.w.run(SafeRunnable.java:23)
E/AndroidRuntime(15676): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:390)
E/AndroidRuntime(15676): 	at java.util.concurrent.FutureTask.run(FutureTask.java:234)
E/AndroidRuntime(15676): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:153)
E/AndroidRuntime(15676): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:267)
E/AndroidRuntime(15676): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
E/AndroidRuntime(15676): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
E/AndroidRuntime(15676): 	at java.lang.Thread.run(Thread.java:856)
E/AndroidRuntime(15676): Caused by: java.lang.IllegalStateException: Lock pixels error, frame buffer allocation failed
E/AndroidRuntime(15676): 	at pl.droidsonroids.gif.GifInfoHandle.renderFrame(Native Method)
E/AndroidRuntime(15676): 	at pl.droidsonroids.gif.GifInfoHandle.g(GifInfoHandle.java:106)
E/AndroidRuntime(15676): 	at pl.droidsonroids.gif.f.rh(GifDrawable.java:92)
E/AndroidRuntime(15676): 	at pl.droidsonroids.gif.w.run(SafeRunnable.java:18)
E/AndroidRuntime(15676): 	... 7 more
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
D/qdmemalloc(  220): ion: Mapped buffer base:0x4623b000 size:1835008 fd:39
D/qdmemalloc(  220): ion: Mapped buffer base:0x400de000 size:4096 fd:69
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
D/qdmemalloc(  220): ion: unmap_buffer base:0x0 size:1835008 offset:0 fd:39
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
D/qdmemalloc(  220): ion: unmap_buffer base:0x0 size:4096 offset:0 fd:69
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
E/OpenGLRenderer(15676): Cannot generate texture from bitmap
D/dalvikvm(  738): GC_FOR_ALLOC freed 7360K, 18% free 65559K/79732K, paused 142ms, total 142ms
```

**Comment by koral--** [comment]
That state:
* message from OpenGLRenderer `Cannot generate texture from bitmap` printed to the log
* `ANDROID_BITMAP_RESULT_ALLOCATION_FAILED` returned by `AndroidBitmap_lockPixels`
occurs randomly when using bitmap with hardware accelerated `Canvas`.
I found no real solution (except disabling hardware acceleration or changing bitmap to something else), previous workaround was to retry at most 20 times with delay and then give up throwing `IllegalStateException` which you have mentioned. New one is to render current (unchanged) frame and schedule retry. That flow is repeated until success or drawable becomes invisible and/or not running. 
If some more elegant solution is discovered I'll replace current workaround.

**Comment by XavierLi** [comment]
Reproduced again. Android version: 4.2.1.

04-09 12:05:31.875: E/OpenGLRenderer(26896): Cannot generate texture from bitmap
04-09 12:05:31.960: I/wpa_supplicant(700): Receive Command: SIGNAL_POLL
04-09 12:05:32.117: E/AndroidRuntime(26896): FATAL EXCEPTION: pool-3-thread-1
04-09 12:05:32.117: E/AndroidRuntime(26896): java.lang.IllegalStateException: Lock pixels error, frame buffer allocation failed
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at pl.droidsonroids.gif.GifInfoHandle.renderFrame(Native Method)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at pl.droidsonroids.gif.GifInfoHandle.renderFrame(GifInfoHandle.java:114)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at pl.droidsonroids.gif.RenderTask.doWork(RenderTask.java:23)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at pl.droidsonroids.gif.SafeRunnable.run(SafeRunnable.java:18)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:390)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at java.util.concurrent.FutureTask.run(FutureTask.java:234)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:153)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:267)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
04-09 12:05:32.117: E/AndroidRuntime(26896): 	at java.lang.Thread.run(Thread.java:856)

**Comment by koral--** [comment]
This issue is fixed in version 1.1.4-SNAPSHOT. Will be available in 1.1.4 release.

**Comment by koral--** [comment]
Version 1.1.4 has been pushed to maven central repository.
